python/orca/src/bigdl/orca/common.py

```diff
@@ -153,9 +153,39 @@ def get_spark_session():
         return OrcaContext.get_sql_context().sparkSession
 
     @staticmethod
-    def get_ray_context():
+    def get_ray_context(initialize=True):
         from bigdl.orca.ray import RayContext
-        return RayContext.get()
+        return RayContext.get(initialize)
+
+    @staticmethod
+    def core_num():
+        sc = OrcaContext.get_spark_context()
+        if sc.getConf().get('spark.master').startswith('local'):
+            num_cores = OrcaContext.get_local_cores_num(sc)
+        else:
+            if sc.getConf().contains("spark.executor.cores"):
+                num_cores = int(sc.getConf().get("spark.executor.cores"))
+        return num_cores
+
+    @staticmethod
+    def node_num():
+        sc = OrcaContext.get_spark_context()
+        if sc.getConf().get('spark.master').startswith('local'):
+            num_nodes = 1
+        else:
+            if sc.getConf().contains("spark.executor.instances"):
+                num_nodes = int(sc.getConf().get("spark.executor.instances"))
+        return num_nodes
+
+    @staticmethod
+    def get_local_cores_num(sc):
+        import re
+        import multiprocessing
+        local_symbol = re.match(r"local\[(.*)\]", sc.master).group(1)
+        if local_symbol == "*":
+            return multiprocessing.cpu_count()
+        else:
+            return int(local_symbol)
```

Comment:
The cluster node# and core# are determined in `init_orca_context`, yes? We should not duplicate the logic here.

Reply:
Yes, that's duplicated. So are we going to use `OrcaContext.node_num` to replace all `ray_ctx.num_ray_nodes`, not like `self.num_ray_nodes = OrcaContext.node_num( )`?